### ISSUE TITLE
When putenv() has been disabled for security reasons

### ISSUE THREAD

**Issue by piotr-cz** [vlucas/phpdotenv#168]
My host decided for some reason to disable this function.
I can still set `$_ENV` and `$_SERVER` variables, but after `$dotenv->load()` I'm getting a PHP warning.

```
Warning: putenv() has been disabled for security reasons in XXX/vendor/vlucas/phpdotenv/src/Loader.php on line 338
```

I was about to suggest that phpdotenv does a check `function_exists('putenv'))` before using this function, but then It would be harder to find out why the `getenv` function is returning false.

On the other hand I had to disable all PHP warnings in front controller because this one was destroying server response before I even started application - I use variables from dotenv during app bootstrap to set debug mode which decides what to do with error reporting.

What is best strategy to handle this situation?

**Comment by rquadling** [comment]
The alternative I've been looking at is https://github.com/josegonzalez/php-dotenv

I was steered towards this one because of the numbers of downloads on packagist, but for me, the php-dotenv approach (yes, a hyphen is all that is different), seems better. And it supports a better parser for the .env files from my use/looking.

**Comment by piotr-cz** [comment]
Thanks.
Yes, it probably looks like best solution for my case - to specify where to store data
```php
(new Loader('.env'))->parse()->toEnv();
```

**Comment by piotr-cz** [comment]
This has been resolved in https://github.com/vlucas/phpdotenv/commit/58633a061610ffd731d7fbbc8f7f009d14b2e4cf if it comes into a release.

Caveat is that as the `putenv` function is disabled/ not being used, it's not possible to read variables using the `getenv` too, but one must resolve to `$_ENV` or `$_SERVER` globals directly.

Even when following the php.ini directive is set: `variables_order = "EGPCS"`.

**Comment by GrahamCampbell** [comment]
Nothing we can do if you turned that function off.

**Comment by piotr-cz** [comment]
Thanks.
Waiting for new release

**Comment by piotr-cz** [comment]
This is available now in v2.4
Thank you
